internal/block/block_pool.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package block
+
+import (
+	"fmt"
+	"syscall"
+)
+
+// BlockPool handles the creation of blocks as per the user configuration.
+type BlockPool struct {
+	// Channel holding free blocks.
+	blocksCh chan Block
```

Comment:
*nit = Can we rename to blocksChannel or similar like freeBlocksChannel instead to signify its usage and helps in readability.

Reply:
Done.